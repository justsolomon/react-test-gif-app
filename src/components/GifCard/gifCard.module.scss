@import "../../assets/scss/base/variables";

.card {
  display: block;
  text-decoration: none;
  color: $primary-color;
  position: relative;
  border-radius: $border-radius--md;

  &__gif,
  &__container {
    height: 100%;
    width: 100%;
    border-radius: $border-radius--md;
  }

  &__user {
    display: none;
    align-items: center;
    background-color: $bg-dark--tertiary;
    position: absolute;
    bottom: 0;
    width: 100%;
    padding: 0.5rem 0.8rem;

    &-avatar {
      height: 40px;
      width: 35px;
      border-radius: $border-radius--sm;
      margin-right: 0.5rem;
    }

    &-info {
      display: flex;
      flex-direction: column;
      font-weight: $font-weight--bold;
      width: 100%;
    }

    &-displayname,
    &-name {
      width: 75%;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    &-displayname {
      color: $white;
      padding-bottom: 4px;
      font-size: $font-size--base;
    }

    &-name {
      color: $bg-light--primary;
      font-size: $font-size--sm;
    }
  }

  &:hover &__user {
    display: flex;
  }
}
