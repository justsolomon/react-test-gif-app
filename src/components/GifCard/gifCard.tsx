import LazyLoad from "react-lazyload";
import Skeleton from "react-loading-skeleton";
import { Link } from "react-router-dom";
import { IGif } from "types/global";
import styles from "./gifCard.module.scss";

interface IGifCard {
  gif: IGif;
}

const GifCard = ({ gif }: IGifCard): JSX.Element => {
  const { images } = gif;

  return (
    <LazyLoad once placeholder={<Skeleton height={images.original.height} />}>
      <Link to={`/gif/${gif.id}`} className={styles["card"]}>
        <div className={styles["card__container"]}>
          <img
            src={
              images.fixed_height_small?.webp ||
              images.fixed_height_downsampled?.webp ||
              images.fixed_height?.webp ||
              images.fixed_width_small?.webp ||
              images.fixed_width_downsampled?.webp ||
              images.fixed_width?.webp ||
              images.downsized?.url ||
              images.downsized_medium?.url ||
              images.downsized_large?.url ||
              images.original.url
            }
            className={styles["card__gif"]}
            alt={gif.title}
          />
          {gif.user ? (
            <div className={styles["card__user"]}>
              <img
                src={gif.user?.avatar_url}
                className={styles["card__user-avatar"]}
                alt={`${gif.user?.display_name}'s avatar`}
              />
              <div className={styles["card__user-info"]}>
                <p className={styles["card__user-displayname"]}>
                  {gif.user?.display_name}
                </p>
                <p
                  className={styles["card__user-name"]}
                >{`@${gif.user?.username}`}</p>
              </div>
            </div>
          ) : null}
        </div>
      </Link>
    </LazyLoad>
  );
};

export default GifCard;
