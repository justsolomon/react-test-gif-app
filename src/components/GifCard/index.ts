export { default as GifCard } from "./gifCard";
