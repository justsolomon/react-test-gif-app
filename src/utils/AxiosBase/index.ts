import axios from "axios";

// Creates a base instance for all axios based request
const instance = axios.create({
  baseURL: `https://api.giphy.com/v1/`
});

instance.interceptors.request.use((config) => {
  //add api key to query params
  config.params["api_key"] = process.env.REACT_APP_API_KEY;

  return config;
});

export default instance;
