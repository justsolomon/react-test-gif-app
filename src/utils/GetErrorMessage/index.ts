import { AxiosError } from "axios";

const getErrorMessage = (err: AxiosError): string => {
  let errorMessage;
  if (err.response) errorMessage = err.response.data.message;
  else errorMessage = err.message;

  return errorMessage;
};

export default getErrorMessage;
