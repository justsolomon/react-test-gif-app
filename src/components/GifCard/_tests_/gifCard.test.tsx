import { render } from "@testing-library/react";
import { gif } from "utils/testUtils";
import GifCard from "../gifCard";

describe("<GifCard />", () => {
  it("should render the GIF card component properly", () => {
    const { asFragment } = render(<GifCard gif={gif} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
