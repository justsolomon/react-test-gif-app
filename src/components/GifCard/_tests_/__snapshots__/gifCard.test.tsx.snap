// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> should render the GIF card component properly 1`] = `
<DocumentFragment>
  <div
    class="lazyload-wrapper "
  >
    <span>
      <span
        class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
      >
        â€Œ
      </span>
    </span>
  </div>
</DocumentFragment>
`;
